backend/app/views.py

```diff
@@ -90,6 +90,15 @@ def example_id(request, example_id):
 
     return render(request, 'index.html', context)
 
+@api_view(['POST'])
+def add_text(request):
+    """
+    API endpoint for adding a piece of text
+    """
+    new_text_obj = Document(request.data)
```

Comment:
I wonder if we should be more explicit about what we're passing in here as `request.data`. As it stands, we're submitting the JSON object directly into the `Document` instance, which means we need to make sure that the front end formats that data in a shape according to what the model expects. I think I'd rather see the front end shape the data according to what the API layer expects. Would we basically create a dictionary here that explicitly pulls the expected properties out of `request.data` and sets them as kwargs on `Document`?

Reply:
Currently, frontend passes a dictionary as `request.data` and then `**request.data` should set it to kwargs that can be directly passed into `Document`.

Reply:
So if I follow correctly, that means that we need to be consistent between the expected kwargs on `Document` and the properties on the JavaScript object we create on the frontend in state to capture user input, right? I think I'd like to avoid that pattern, as we've basically created an explicit dependency between the absolute highest and absolute lowest parts of the stack (in effect, the property on a JavaScript object needs to match the field name in the database). In general, it feels safer to me to have the API layer be the 'translator', so to speak, between the JSON object and the models. In this case, the properties on the JavaScript state object are very simple, but in other contexts it might be more complex. It's obviously absurdly verbose to manually create a dictionary out of `request.data`, since that's what the `api_view` decorator does anyway, but doing so would create the pattern that the API layer handles translation between the front end JavaScript and the model and has the added benefit of making it explicit which properties the API expects that data to contain. Does that all seem legit?

Reply:
Yes, this sounds legit, I'll update the API view